src/consensus/blockchain.py

```diff
@@ -253,16 +246,20 @@ async def receive_block(
 
         sub_block = block_to_sub_block_record(
             self.constants,
-            self.sub_blocks,
-            self.sub_height_to_hash,
+            self,
             required_iters,
             block,
             None,
         )
 
         # Always add the block to the database
         await self.block_store.add_full_block(block, sub_block)
-        self.sub_blocks[sub_block.header_hash] = sub_block
+
+        self.__sub_blocks[sub_block.header_hash] = sub_block
+        if sub_block.sub_block_height not in self.__sub_heights_in_cache.keys():
+            self.__sub_heights_in_cache[sub_block.sub_block_height] = []
+        self.__sub_heights_in_cache[sub_block.sub_block_height].append(sub_block.header_hash)
+        self.clean_sub_block_record(sub_block.sub_block_height - self.constants.SUB_BLOCKS_CACHE_SIZE)
```

Comment:
We really need to make sure to check that we are not syncing, before calling receive block, because we don't want to clean any sub-block records that are neede for syncing

Reply:
good point, i moved the clean up to full_node.
we call cleanup after receiving a sub block in respond_sub_block only if we are not in sync mode, 
and in the _sync method finally clause